File: octave/c001e.m
% Solution to /r/dailyprogrammer Challenge #1 Easy
% Jimmy Nguyen
%
% Prompt:
%     Create a program that will ask the user's name, age, and reddit username.
% Have it tell them the information back, in the format:
% =============================================================================
% your name is (blank), you are (blank) years old, and your username is (blank)
% =============================================================================
%
% Bonus:
%     For extra credit, have the program log this information in a file to be
% accessed later.
%
% Link to challenge:
%     https://www.reddit.com/r/dailyprogrammer/comments/pih8x/easy_challenge_1/
function output = c001e(varargin)
	if nargin < 3
		[name, age, username] = prompt_user();
	else
		[name, age, username] = deal(varargin{1}, varargin{2}, varargin{3});
	end
	output = print_and_save(name, age, username);

function [name, age, username] = prompt_user()
	name     = input('Name: ', 's');
	age      = input('Age : ', 's');
	username = input('Reddit Username: ', 's');

function output = print_and_save(name, age, username)
	output = sprintf(['your name is %s, you are %s years old, and your ', ...
			'username is %s\n'], name, age, username);
	fprintf('\n%s\n', output);
	save_to_file(output);

function save_to_file(output)
	file_path = '../trash/info.log';
	fh = fopen(file_path, 'a');
	fprintf(fh, output);
	fclose(fh);
	fprintf(['Your information has been saved in the file "%s"\n'], file_path);

%!assert(c001e('NAME', 'AGE', 'USERNAME'), ['your name is NAME, you are AGE years old, and your username is USERNAME', sprintf('\n')])

File: challenges/easy/c001e/java/c001e.java
/*
 * Solution to /r/dailyprogrammer Challenge #1 Easy
 * Jimmy Nguyen
 *
 * Prompt:
 *     Create a program that will ask the user's name, age, and reddit username.
 * Have it tell them the information back, in the format:
 * =============================================================================
 * your name is (blank), you are (blank) years old, and your username is (blank)
 * =============================================================================
 *
 * Bonus:
 *     For extra credit, have the program log this information in a file to be
 * accessed later.
 *
 * Link to challenge:
 *     https://www.reddit.com/r/dailyprogrammer/comments/pih8x/easy_challenge_1/
 */
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public final class c001e {
	private static final String FILE_PATH = "../../trash/info.log";
	private static Scanner scanner;

	private static String input(String prompt) {
		System.out.print(prompt);
		return scanner.nextLine();
	}

	private static void save_to_file(String output) {
		BufferedWriter bufferedWriter = null;
		try {
			bufferedWriter = new BufferedWriter(new FileWriter(FILE_PATH, true));
			bufferedWriter.write(output);
			bufferedWriter.newLine();
			bufferedWriter.flush();
			System.out.printf("Your information has been saved in the file \"%s\" in the current directory\n", FILE_PATH);
		} catch (IOException ioe) {
			ioe.printStackTrace();
		} finally {
			if (bufferedWriter != null) {
				try {
					bufferedWriter.close();
				} catch (IOException ioe) {
					ioe.printStackTrace();
				}
			}
		}
	}

	public static void main(String[] args) {
		scanner = new Scanner(System.in);

		// ask for the user's information
		String name     = input("Name: ");
		String age      = input("Age : ");
		String username = input("Reddit Username: ");

		// format the output string
		String output = String.format("your name is %s, you are %s years old, and your username is %s", name, age, username);

		// print the output string
		System.out.println("\n" + output + "\n");

		// save the output string to a file
		save_to_file(output);
	}
}
